### PR TITLE
add support for filtering assets page using evm chain & address

### PR DESCRIPTION
**Pull request by prettyirrelevant**
Support for assets filtering using `evm_chain` & `address`

### CODE REVIEW

File: rotkehlchen/api/v1/schemas.py [hunk 1]
```diff
@@ -1880,10 +1880,15 @@ class IgnoredAssetsHandling(SerializableEnumMixin):
     SHOW_ONLY = auto()
 
 
-class AssetsPostSchema(DBPaginationSchema, DBOrderBySchema):
+class AssetsPostSchema(DBPaginationSchema, DBOrderBySchema, OptionalEthereumAddressSchema):
```
Comment (yabirgb) [inline rotkehlchen/api/v1/schemas.py]: if you defined the fields then this `OptionalEthereumAddressSchema` is not needed. Indeed the type fields are different

File: rotkehlchen/tests/api/test_assets.py [hunk 2]
```diff
@@ -499,6 +499,51 @@ def test_get_all_assets(rotkehlchen_api_server):
...
+        status_code=HTTPStatus.BAD_REQUEST,
+    )
+
+    # check that filtering by evm_chain works.
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'allassetsresource',
+        ),
+        json={'evm_chain': 'ethereum', 'limit': 10, 'offset': 0},
```
Comment (yabirgb) [inline rotkehlchen/tests/api/test_assets.py:522]: can you filter by symbol and `evm_chain` removing the limit? As it is now could be that you are returning the results with chain ethereum first and there are more that due the limit are not shown

File: docs/api.rst [hunk 3]
```diff
@@ -3057,6 +3057,8 @@ Querying all supported assets
    :reqjson string name: The name of asset to be used to filter the result data. Optional.
    :reqjson string symbol: An asset symbol to be used to filter the result data. Optional.
    :reqjson string asset_type: The category of an asset to be used to filter the result data. Optional.
+   :reqjson string evm_chain: The name for the evm chain of the asset(s) to be used to filter the result data. Possible values are ``ethereum``, ``optimism``, etc. Optional.
+   :reqjson string address: The address of the evm asset to be used to filter the result data. Optional.
    :reqjson bool show_user_owned_assets_only: A flag to specify if only user owned assets should be returned. Defaults to ``"false"``. Optional.
```
Comment (yabirgb) [inline docs/api.rst:3062]: Can you add to the docs the constraint that if chain is set then asset type can only be evm token?

File: rotkehlchen/db/filtering.py [hunk 4]
```diff
@@ -1073,12 +1073,13 @@ def make(
             offset: Optional[int] = None,
             name: Optional[str] = None,
             symbol: Optional[str] = None,
+            evm_chain: Optional[ChainID] = None,
```
[the remark this answers is not in the document]
Comment (prettyirrelevant) [reply rotkehlchen/db/filtering.py:1076]: thanks, was mixing up different domains

**Review by yabirgb — APPROVED** [review]
Failing tests are the ones fixes in bugfixes

File: rotkehlchen/api/v1/schemas.py [hunk 5]
```diff
@@ -1905,6 +1910,15 @@ def validate_schema(
                 field_name='order_by_attributes',
             )
 
+        if (
+            data['evm_chain'] is not None and
+            data['asset_type'] not in (None, AssetType.EVM_TOKEN)
+        ):
+            raise ValidationError(
+                message='Filtering by evm_chain is only supported by evm tokens',
```
Comment (LefterisJP) [inline rotkehlchen/api/v1/schemas.py:1918]:
```suggestion
                message='Filtering by evm_chain is only supported for evm tokens',
```

File: rotkehlchen/tests/api/test_assets.py [hunk 6]
```diff
@@ -499,6 +499,49 @@ def test_get_all_assets(rotkehlchen_api_server):
...
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'allassetsresource',
+        ),
+        json={'asset_type': 'fiat', 'evm_chain': 'ethereum'},
+    )
+    assert_error_response(
+        response=response,
+        contained_in_msg='Filtering by evm_chain is only supported by evm tokens',
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_assets.py:512]:
```suggestion
        contained_in_msg='Filtering by evm_chain is only supported for evm tokens',
```

**Review by LefterisJP — APPROVED** [review]
I left a small nitpick that should be addressed in a subsequent PR. Please **remember** to do it @prettyirrelevant

**Pull request merged**